### PR TITLE
Adding Small(ish) Bench Tests to Auto Test List

### PR DESCRIPTION
**Pull request by simpsont-oci**

### CODE REVIEW

File: performance-tests/bench_2/test_controller/main.cpp [hunk 1]
```diff
@@ -515,4 +518,12 @@ void handle_reports(const std::vector<Bench::WorkerReport>& parsed_reports, std:
   result_out << std::endl;
 
   consolidated_round_trip_jitter_stats.pretty_print(result_out, "round trip jitter");
+
+  if (total_undermatched_readers ||
+      total_undermatched_writers ||
+      total_out_of_order_data_count ||
+      total_duplicate_data_count) {
+    result = -1;
+  }
```
Comment (iguessthislldo) [inline performance-tests/bench_2/test_controller/main.cpp]: Please don't return a negative number from main: #1362

```suggestion
  result = total_undermatched_readers ||
    total_undermatched_writers ||
    total_out_of_order_data_count ||
    total_duplicate_data_count;
```

File: bin/dcps_tests.lst [hunk 2]
```diff
@@ -409,3 +409,9 @@ tests/DCPS/RtpsRelay/Smoke/run_test.pl single: !DCPS_MIN CXX11 RTPS !NO_BUILT_IN
 tests/DCPS/RtpsRelay/STUN/run_test.pl: !DCPS_MIN CXX11 RTPS !NO_BUILT_IN_TOPICS !OPENDDS_SAFETY_PROFILE !IPV6
 
 tests/DCPS/ParticipantLocationTopic/run_test.pl noice: !DCPS_MIN CXX11 RTPS !NO_BUILT_IN_TOPICS !OPENDDS_SAFETY_PROFILE !IPV6
+
+performance-tests/bench_2/run_test.pl disco: !DCPS_MIN !NO_MCAST RTPS !DDS_NO_OWNERSHIP_PROFILE! CXX11 RTPS
```
Comment (iguessthislldo) [inline bin/dcps_tests.lst]:
```suggestion
performance-tests/bench_2/run_test.pl disco: !DCPS_MIN !NO_MCAST RTPS !DDS_NO_OWNERSHIP_PROFILE CXX11 RTPS
```

And the rest?

**Comment by mitza-oci** [comment]
Does this need to be resolved in this PR? https://github.com/objectcomputing/OpenDDS/blob/3387c9655bd9306d6e747e5302e219f880dc49cb/performance-tests/bench_2/node_controller/main.cpp#L146

**Comment by simpsont-oci** [comment]
> Does this need to be resolved in this PR?
> 
> https://github.com/objectcomputing/OpenDDS/blob/3387c9655bd9306d6e747e5302e219f880dc49cb/performance-tests/bench_2/node_controller/main.cpp#L146

It's rare for the node_controller to segfault in my windows testing so far. A bigger concern might be that killing the Perl script doesn't correctly kill the node_controller & worker processes (compare with bash scripts, which kills sub-processes). That doesn't seem like a show-stopper though, since most of the time we'll want the whole test to run to completion.

File: performance-tests/bench_2/test_controller/main.cpp [hunk 3]
```diff
@@ -357,6 +357,7 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[])
       if (reports.size() != allocated_scenario.expected_reports) {
         result_file << "ERROR: Only received " << reports.size() << "out of " << allocated_scenario.expected_reports << " valid reports!" << std::endl;
         std::cerr << "ERROR: Only received " << reports.size() << " out of " << allocated_scenario.expected_reports << " valid reports!" << std::endl;
+	result = 1;
```
Comment (iguessthislldo) [inline performance-tests/bench_2/test_controller/main.cpp]:
```suggestion
	  result = 1;
```

File: bin/dcps_tests.lst [hunk 4]
```diff
@@ -409,3 +409,9 @@ tests/DCPS/RtpsRelay/Smoke/run_test.pl single: !DCPS_MIN CXX11 RTPS !NO_BUILT_IN
 tests/DCPS/RtpsRelay/STUN/run_test.pl: !DCPS_MIN CXX11 RTPS !NO_BUILT_IN_TOPICS !OPENDDS_SAFETY_PROFILE !IPV6
 
 tests/DCPS/ParticipantLocationTopic/run_test.pl noice: !DCPS_MIN CXX11 RTPS !NO_BUILT_IN_TOPICS !OPENDDS_SAFETY_PROFILE !IPV6
+
+performance-tests/bench_2/run_test.pl disco: !DCPS_MIN !NO_MCAST RTPS !DDS_NO_OWNERSHIP_PROFILE CXX11 RTPS
```
Comment (mitza-oci) [inline bin/dcps_tests.lst]:
```suggestion
performance-tests/bench_2/run_test.pl disco: !DCPS_MIN !NO_MCAST RTPS !DDS_NO_OWNERSHIP_PROFILE CXX11
```

File: performance-tests/bench_2/node_controller/main.cpp [hunk 5]
```diff
@@ -116,17 +116,17 @@ class Worker {
     }
   }
 
-  ACE_Process_Options get_proc_opts() const
+  std::shared_ptr<ACE_Process_Options> get_proc_opts() const
   {
-    ACE_Process_Options proc_opts;
+    std::shared_ptr<ACE_Process_Options> proc_opts(new ACE_Process_Options());
```
Comment (mitza-oci) [inline performance-tests/bench_2/node_controller/main.cpp]: Use `make_shared`?

File: performance-tests/bench_2/set_bench_env.cmd [hunk 6]
```diff
@@ -0,0 +1,3 @@
+set BENCH_ROOT=%DDS_ROOT%\performance-tests\bench_2
+set BENCH_BUILDER_ROOT=%DDS_ROOT%\performance-tests\bench_2
+set PATH=%PATH%:%PATH%;%BENCH_BUILDER_ROOT%\lib;%BENCH_ROOT%\lib
```
Comment (mitza-oci) [inline performance-tests/bench_2/set_bench_env.cmd]:
```suggestion
set PATH=%PATH%;%BENCH_BUILDER_ROOT%\lib;%BENCH_ROOT%\lib
```

File: performance-tests/bench_2/test_controller/main.cpp [hunk 7]
```diff
@@ -356,6 +357,7 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[])
       if (reports.size() != allocated_scenario.expected_reports) {
         result_file << "ERROR: Only received " << reports.size() << "out of " << allocated_scenario.expected_reports << " valid reports!" << std::endl;
         std::cerr << "ERROR: Only received " << reports.size() << " out of " << allocated_scenario.expected_reports << " valid reports!" << std::endl;
+        result = 1;
```
Comment (mitza-oci) [inline performance-tests/bench_2/test_controller/main.cpp]:
```suggestion
        result = EXIT_FAILURE;
```

File: performance-tests/bench_2/test_controller/main.cpp [hunk 8]
```diff
@@ -35,10 +35,11 @@ using namespace Bench::TestController;
 
 std::string bench_root;
 
-void handle_reports(const std::vector<Bench::WorkerReport>& parsed_reports, std::ostringstream& result_out);
+int handle_reports(const std::vector<Bench::WorkerReport>& parsed_reports, std::ostringstream& result_out);
 
 int ACE_TMAIN(int argc, ACE_TCHAR* argv[])
 {
+  int result = 0;
```
Comment (mitza-oci) [inline performance-tests/bench_2/test_controller/main.cpp]:
```suggestion
  int result = EXIT_SUCCESS;
```

File: performance-tests/bench_2/test_controller/main.cpp [hunk 9]
```diff
@@ -365,11 +367,12 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[])
...
   std::cout << "Finished" << std::endl;
 
-  return 0;
+  return result;
 };
 
-void handle_reports(const std::vector<Bench::WorkerReport>& parsed_reports, std::ostringstream& result_out)
+int handle_reports(const std::vector<Bench::WorkerReport>& parsed_reports, std::ostringstream& result_out)
 {
+  int result = 0;
```
Comment (mitza-oci) [inline performance-tests/bench_2/test_controller/main.cpp]: same thing as above...

File: performance-tests/bench_2/worker/main.cpp [hunk 10]
```diff
@@ -217,15 +217,14 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[]) {
...
-        while (cv.wait_for(lock, dur) != std::cv_status::timeout) {}
+        timeout_time = std::chrono::high_resolution_clock::now() - get_duration(config.create_time);
       } else {
-        auto timeout_time = std::chrono::system_clock::time_point(get_duration(config.create_time));
-        std::unique_lock<std::mutex> lock(cv_mutex);
-        while (cv.wait_until(lock, timeout_time) != std::cv_status::timeout) {}
+        timeout_time = std::chrono::high_resolution_clock::time_point(get_duration(config.create_time));
       }
+      std::unique_lock<std::mutex> lock(cv_mutex);
+      while (cv.wait_until(lock, timeout_time) != std::cv_status::timeout) {}
```
Comment (mitza-oci) [inline performance-tests/bench_2/worker/main.cpp]: What is it actually waiting for?  The condition variable wait can't complete until some external state change, protected by the cv's mutex, has been observed.

File: bin/dcps_tests.lst (hunk 4 above)
Comment (mitza-oci) [reply bin/dcps_tests.lst]: similar idea on a few lines below...

Comment (simpsont-oci) [reply bin/dcps_tests.lst]: They're getting run. The fan_frag test fails sometimes on the PR builder, which we'll want to look into, but I don't think it's the test's fault so I'm OK with that for now. I'm also curious to see if there are any platform / feature specific failures that we'll want to focus on first (from the scoreboard), which is why I haven't done a deep dive there yet. Just trying to get the tests in place first.

File: performance-tests/bench_2/worker/main.cpp (hunk 10 above)
Comment (simpsont-oci) [reply performance-tests/bench_2/worker/main.cpp]: It's waiting for me to remember / look up how to just do a normal thread sleep in C++11. :)

**Pull request merged**